### PR TITLE
Move semaphore tests from api to extensions

### PR DESCRIPTION
**Pull request by pj87**
Moved cl_khr_semaphore tests from core api tests to a separate extension test as suggested in
https://github.com/KhronosGroup/OpenCL-CTS/issues/1588 .

### CODE REVIEW

File: test_conformance/extensions/cl_khr_semaphore/main.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#include "procs.h"
+#include "harness/testHarness.h"
+
+#if !defined(_WIN32)
+#include <unistd.h>
+#endif
+
+// FIXME: To use certain functions in harness/imageHelpers.h
+// (for example, generate_random_image_data()), the tests are required to
+// declare the following variables (<rdar://problem/11111245>):
```
Comment (EwanC) [inline test_conformance/extensions/cl_khr_semaphore/main.cpp:30]: Don't think this comment from api/main.cpp applies here, can remove

**Review by EwanC — APPROVED** [review]
Looks good to me (one very minor comment). Mentioning issue https://github.com/KhronosGroup/OpenCL-CTS/issues/1624 which can be closed once this is merged

**Review by EwanC — APPROVED** [review]

**Comment by lakshmih** [comment]
Reviewing. Thanks.

**Comment by bashbaug** [comment]
Merging as discussed in the March 7th teleconference.

**Pull request merged**